### PR TITLE
feat(frontend): save carousel hidden slides to backend

### PR DESCRIPTION
**Pull request by AntonioVentilii**
# Motivation

We use the backend method `add_hidden_dapp_id` introduced with PR #3777  , to save the hidden slides/dApps in the backend.

# Changes

- Add event `oisyRefreshUserProfile` to refresh the user profile in `LoaderUserProfile`. It is useful to emit it when we know that we change the user profile.
- Add method `add_hidden_dapp_id` in `DappCarousel` when closing a slide.

# Tests

https://github.com/user-attachments/assets/c849ed22-dfa3-4f51-90e8-5d6a446b5dfb

### CODE REVIEW

File: src/frontend/src/lib/components/loaders/LoaderUserProfile.svelte [hunk 1]
```diff
@@ -14,6 +14,11 @@
 	};
 
 	$: $authIdentity, load({});
+	const reload = () => {
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/loaders/LoaderUserProfile.svelte:17]:
```suggestion
	const reload = () => {
```

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte [hunk 2]
```diff
@@ -30,6 +36,18 @@
 		if (idx !== -1) {
 			carousel.removeSlide(idx);
 		}
+
+		if (isNullish($authIdentity) || isNullish($userProfileStore)) {
```
Comment (DenysKarmazynDFINITY) [inline src/frontend/src/lib/components/dapps/DappsCarousel.svelte:40]: For `isNullish($authIdentity)` we can use `await nullishSignOut();`.

**Review by DenysKarmazynDFINITY — APPROVED** [review]
Small comment, otherwise LGTM, thanks!

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte (hunk 2 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/components/dapps/DappsCarousel.svelte]: Good one!!! Included!

**Pull request merged**